packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-sign-in-soc-idp/java/integrationsteps.md

```diff
@@ -1,3 +1,57 @@
 ## Integration steps
 
-The Java SDK integration steps for this use case are coming soon. See the [Okta Identity Engine Java SDK usage guide](https://github.com/okta/okta-idx-java#usage-guide) for more information.
+### Step 1: Sign in with Facebook
+
+Use [IDXAuthenticationWrapper](https://github.com/okta/okta-idx-java/blob/master/api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java) to start the sign-in process with Okta when the user goes to the sign-in page.
+
+```java
+AuthenticationResponse beginResponse = idxAuthenticationWrapper.begin()
+```
+
+Use the [AuthenticationResponse.getIdps()](https://github.com/okta/okta-idx-java/blob/master/api/src/main/java/com/okta/idx/sdk/api/response/AuthenticationResponse.java#L91) function to return a list of social Identity Provider (IdP) options configured in your org's routing rule.
+
+```java
+List<Idp> idps = authenticationResponse.getIdps();
+```
+
+You need to build a generic sign-in view page with the social sign-in options available from the list of IdPs returned. However, in this use case, Facebook is the only configured IdP in the list.
+
+```html
+<div th:case="'FACEBOOK'">
+    <a th:href="@{${idp.href}}" class="btn btn-lg btn-social btn-facebook" id="btn-facebook">
+        <i class="fa fa-facebook fa-fw"></i> Login with Facebook
+    </a>
+</div>
+```
+
+The following example displays the **Login with Facebook** button.
+
+<div class="common-image-format">
+
+![Social sign-in screenshot for Java](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-social-sign-in-link-java.png
+ "Social sign-in screenshot for Java")
+
+</div>
+
+### Step 2: User signs in with Facebook
+
+When the user selects the **Login with Facebook** option, they are directed to the Facebook sign-in page.
+
+After the user signs in to Facebook successfully, Facebook routes the user to the location specified in **Valid OAuth Redirect URIs** from the Facebook developer site.
+
+The **Valid OAuth Redirect URIs** for your Okta org is in the format: `https://{yourOktaDomain}/oauth2/v1/authorize/callback`.
+
+> **Note:** See [Step 1: Create a Facebook app in Facebook](/docs/guides/oie-embedded-common-org-setup/java/main/#step-1-create-a-facebook-app-in-facebook) for details on configuring the **Valid OAuth Redirect URIs** value.
+
+### Step 3: Handle the callback from Okta
+
+Okta returns the Interaction code in the callback to the **Sign-in redirect URI** location specified in the [Create new application](/docs/guides/oie-embedded-common-org-setup/java/main/#step-4-create-new-application) section. You need to handle the callback by exchanging the Interaction code for token.
```

Comment:
@vanngo-okta I can understand the wish to avoid "see here" type language related to links; thanks @susanharper-okta . but I still think "Interaction code for token." should really grammatically awkward. How about "Interaction code for a token.", if we still want to keep it generic?

Reply:
Yes, I'll add the "a" :-)